File: 001_two_sums.cpp
/**
* * Question Link: https://leetcode.com/problems/two-sum/
* * Sample Input:  nums = [2,7,11,15], target = 9
* * Sample Output: Output: [0,1]
*/

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int sum = 0;
        vector<int> ans;
        for ( int i=0; i<nums.size(); i++){
            for ( int j=i+1; j<nums.size(); j++ ){
                if ( (nums[i]+nums[j]) == target){
                    ans.push_back(i);
                    ans.push_back(j);
                }
            }
        }
        return ans;
    }
};

File: 001_two_sums.ts
interface Dict {
    [num: number]: number[];
}

function twoSum(nums: number[], target: number): number[] {
    const dict: Dict = {};
    let result: number[] = [];
    for (let i = 0; i < nums.length; i++) {
        let temp = nums[i];
        if (!(temp in dict)) {
            dict[temp] = [i];
        } else dict[temp].push(i);
    }

    Object.keys(dict).forEach((value) => {
        const x = Number.parseInt(value);
        const y = target - x;
        if (y in dict) {
            if (x === y) {
                result = [dict[x][0], dict[x][1]];
            } else {
                result = [dict[x][0], dict[y][0]];
            }
        }
    });

    return result;
}
